services/datamanager/builtin/file_deletion.go

```diff
@@ -0,0 +1,134 @@
+package builtin
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ricochet2200/go-disk-usage/du"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/services/datamanager/datacapture"
+	"go.viam.com/rdk/services/datamanager/datasync"
+)
+
+// TODO change these values back to what they should be.
+var (
+	fsThresholdToTriggerDeletion = .95
+	captureDirToFSUsageRatio     = .5
+	deleteEveryNth               = 4
+)
+
+var errAtSizeThreshold = errors.New("capture dir is at correct size")
+
+func shouldDeleteBasedOnDiskUsage(ctx context.Context, captureDirPath string, logger logging.Logger) (bool, error) {
+	usage := du.NewDiskUsage(captureDirPath)
+	// we get usage this way to ensure we get the amount of remaining space in the partition.
+	// calling usage.Usage() returns the usage of the whole disk, not the user partition
+	usedSpace := 1.0 - float64(usage.Available())/float64(usage.Size())
+	if math.IsNaN(usedSpace) {
+		return false, nil
+	}
+	if usedSpace < fsThresholdToTriggerDeletion {
+		logger.Warnf("disk not full enough, exiting used space: %f", usedSpace)
+		return false, nil
+	}
+	// Walk the dir to get capture stats
+	return exceedsDeletionThreshold(ctx, captureDirPath, float64(usage.Size()), logger)
+}
+
+func exceedsDeletionThreshold(ctx context.Context, captureDirPath string, fsSize float64, logger logging.Logger) (bool, error) {
+	var dirSize int64
+
+	readSize := func(path string, d fs.DirEntry, err error) error {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+
+		if !d.IsDir() {
+			fileInfo, err := d.Info()
+			if err != nil {
+				return err
+			}
+			dirSize += fileInfo.Size()
+			if float64(dirSize)/fsSize >= captureDirToFSUsageRatio {
+				logger.Warnf("At threshold to delete, going to delete usage ratio %f threshold %d", float64(dirSize)/fsSize, captureDirToFSUsageRatio)
+				return errAtSizeThreshold
+			}
+		}
+		return nil
+	}
+
+	err := filepath.WalkDir(captureDirPath, readSize)
+	if err != nil {
+		if errors.Is(err, errAtSizeThreshold) {
+			return true, nil
+		}
+		return false, err
+	}
+	logger.Warnf("Not at threshold size %f threshold %f", float64(dirSize)/fsSize, captureDirToFSUsageRatio)
```

Comment:
[nit] Same comment here about readability, to make it clear what we mean — that the threshold of directory size to file system size is currently X but we need to hit the threshold Y

Also perhaps should move this into debug, since otherwise will spam the logs in the default case that there's nothing to delete?